splunk_add_on_ucc_framework/dashboard.py

```diff
@@ -136,6 +143,37 @@ def generate_dashboard(
     else:
         panels = global_config.dashboard.get("panels", [])
         panel_names = [panel["name"] for panel in panels]
-        content = generate_dashboard_content(addon_name, panel_names)
+        custom_components = ""
+        if PANEL_CUSTOM in panel_names:
+            dashboard_components_path = os.path.abspath(
+                os.path.join(
+                    global_config.original_path,
+                    os.pardir,
+                    "dashboard_components.xml",
+                )
+            )
+            try:
+                custom_xml = ElementTree.parse(dashboard_components_path)
+            except FileNotFoundError:
+                logger.error(
+                    f"Custom dashboard page set in globalConfig.json but "
+                    f"file {dashboard_components_path} not found"
+                )
+                sys.exit(1)
+            except ElementTree.ParseError:
+                logger.error(f"{dashboard_components_path} it's not a valid xml file")
+                sys.exit(1)
+            else:
+                for row in custom_xml.findall("row"):
```

Comment:
Can we do the same trick with `custom-dashboard` tag and not `row`?

Reply:
You mean to replace `row` with `custom-dashboard` ? We could use `custom-dashboard` instead of `row` but finally we will need place this custom content inside `row` tag and remove `custom-dashboard` so I wonder if its worth it.

Reply:
Sorry, my comment was not very useful.

Can we use `custom-dashboard` as a root tag and not do `custom_xml.findall("row"):` - but simply read all the content inside `custom-dashboard` tag without relying on `row` element to be present?

Reply:
Okay, let's leave it as it is, let's update the documentation with a link to `panel` and `row` (https://docs.splunk.com/Documentation/Splunk/latest/Viz/PanelreferenceforSimplifiedXML#panel).

Reply:
Let's do 2 more validations:

* make sure that the root element is `custom-dashboard`, if not - throw an error and stop the build
* make sure that all the child elements of `custom-dashboard` are `row` elements, if not - throw an error and stop the build